app/experimenter/experiments/serializers/design.py

```diff
@@ -170,6 +170,20 @@ class Meta(ExperimentDesignVariantBaseSerializer.Meta):
         model = ExperimentVariant
 
     def validate_preferences(self, data):
+        variant = None
+        error_list = [{"pref_name": "You cannot swap pref names."}] * len(data)
+        for pref in data:
+            id = pref.get("id")
+            if id and not variant:
+                variant = VariantPreferences.objects.get(id=id).variant
+            if (
+                variant
```

Comment:
If you look up the variant at the top, you can move

```py
if variant:
```

to the top before the loop